ui/raidboss/data/05-shb/raid/e12s.js

```diff
@@ -351,6 +389,165 @@ export default {
+    },
+    {
+      id: 'E12S Relativity Debuff Collector',
+      // 690 Spell-In-Waiting: Flare
+      // 996 Spell-In-Waiting: Unholy Darkness
+      // 998 Spell-In-Waiting: Shadoweye
+      // 99C Spell-In-Waiting: Dark Eruption
+      // 99E Spell-In-Waiting: Dark Blizzard III
+      // 99F Spell-In-Waiting: Dark Aero III
+      netRegex: NetRegexes.gainsEffect({ effectId: ['690', '99[68CEF]'] }),
+      condition: (data, matches) => data.phase === 'intermediate' && matches.target === data.me,
+      preRun: (data, matches) => {
+        data.debuffs = data.debuffs || {};
+        data.debuffs[matches.effectId.toUpperCase()] = parseFloat(matches.duration);
+      },
+      infoText: (data, _, output) => {
+        const unsortedIds = Object.keys(data.debuffs);
+        if (unsortedIds.length !== 3)
+          return;
+
+        // Sort effect ids descending by duration.
+        const sortedIds = unsortedIds.sort((a, b) => data.debuffs[b] - data.debuffs[a]);
+        const keys = sortedIds.map((effectId) => effectIdToOutputStringKey[effectId]);
+
+        // Stash outputstring keys to use later.
+        data.intermediateDebuffs = [keys[1], keys[2]];
+
+        return output.comboText({
+          effect1: output[keys[0]](),
+          effect2: output[keys[1]](),
+          effect3: output[keys[2]](),
+        });
+      },
+      outputStrings: Object.assign({
+        comboText: {
+          en: '${effect1} > ${effect2} > ${effect3}',
+        },
+      }, intermediateRelativityOutputStrings),
+    },
+    {
+      id: 'E12S Relativity Debuffs',
+      // Players originally get `Spell-in-Waiting: Return` or `Spell-in-Waiting: Return IV`.
+      // When Spell-in-Waiting Return IV wears off, players get Return IV effect.
+      // When Return IV effect wears off, players get Return effect.
+      // When Return effect wears off, players go back to previous locations
+      //
+      // Return = 994
+      // Return IV = 995
+      netRegex: NetRegexes.gainsEffect({ effectId: '99[45]' }),
+      condition: Conditions.targetIsYou(),
+      response: (data, _, output) => {
+        // cactbot-builtin-response
+        output.responseOutputStrings = Object.assign({
+          moveAway: {
+            en: 'Move!',
+            de: 'Bewegen!',
+            fr: 'Bougez !',
+            ja: '避けて！',
+            cn: '快躲开！',
+            ko: '이동하기!',
+          },
+        }, intermediateRelativityOutputStrings);
+
+        if (data.phase !== 'intermediate')
+          return { infoText: output.moveAway() };
+
+        const key = data.intermediateDebuffs && data.intermediateDebuffs.shift();
+        if (!key)
+          return { infoText: output.moveAway() };
+        return { alertText: output[key]() };
+      },
+    },
+    {
+      // For all relativities, players should look outside during the final return effect.
+      id: 'E12S Relativity Look Outside',
+      netRegex: NetRegexes.gainsEffect({ effectId: '994' }),
```

Comment:
This is incorrectly triggered during Basic Relativity. Instead, for Basic Relativity it should be a few seconds before the Spell-in-Waiting: Return (9A0) ends.